src/Parser.js

```diff
@@ -94,6 +94,25 @@ export function parse (value, root, parent, rule, rules, rulesets, pseudo, point
 
 				index = offset = property = 0, variable = ampersand = 1, type = characters = '', length = pseudo
 				break
+			// \
+			case 92:
+				if (from(peek()).match(/[^a-f0-9]/i)) {
```

Comment:
This looks overly involved, why not just `next()` the `continue` i.e

```js
case 92: next()
    continue
```

Reply:
Because then the tests fail!

```
 2 failing

  1) Parser
       escaped chars in selector identifiers:

      AssertionError: expected '.user.B{color:red;}.user.xampleom{color:blue;}.user.ownerounder{color:green;}' to equal '.user.B\\&W{color:red;}.user.\\@example\\.com{color:blue;}.user.owner\\/founder{color:green;}'
      + expected - actual

      -.user.B{color:red;}.user.xampleom{color:blue;}.user.ownerounder{color:green;}
      +.user.B\&W{color:red;}.user.\@example\.com{color:blue;}.user.owner\/founder{color:green;}
      
      at Context.<anonymous> (test/Parser.js:142:10)
      at processImmediate (node:internal/timers:464:21)

  2) Parser
       escaped hex digits in selector identifiers:

      AssertionError: expected '.user.BW{color:red;}.user.B0026W{color:green;}.user.B W{color:blue;}.user.endsWith00A9 a.childNode{color:green;}.user.endsWith a.childNode{color:yellow;}' to equal '.user.B\\26W{color:red;}.user.B\\000026W{color:green;}.user.B\\26 W{color:blue;}.user.endsWith\\0000A9  a.childNode{color:green;}.user.endsWith\\AE  a.childNode{color:yellow;}'
      + expected - actual

      -.user.BW{color:red;}.user.B0026W{color:green;}.user.B W{color:blue;}.user.endsWith00A9 a.childNode{color:green;}.user.endsWith a.childNode{color:yellow;}
      +.user.B\26W{color:red;}.user.B\000026W{color:green;}.user.B\26 W{color:blue;}.user.endsWith\0000A9  a.childNode{color:green;}.user.endsWith\AE  a.childNode{color:yellow;}
      
      at Context.<anonymous> (test/Parser.js:154:10)
      at processImmediate (node:internal/timers:464:21)
```

The output still needs the `\` and the next char, so I assume we need a `characters += ` to appear somewhere.

And we sometimes need to preserve double spaces `  ` when they follow hex code sequences, but not when they follow escaped non-hex code characters.

Reply:
>so I assume we need a characters += to appear somewhere.

Yes we can concat the two characters that follow:

```js
case 92: characters += from(next()) + from(next())
    continue
```

Reply:
At a glance - this skips the slash? Would it be more appropriate to do:

```
case 92: characters += from(character) + from(next())
    continue;
```

That still won't pass one of the tests - this solution does nothing for the escaped 6 character hex code followed by a double space? https://jsfiddle.net/8ouxLvdf/

Changing the code, the tests say:

```
1) Parser
       double spaces after escaped hex codes in selector identifiers:

      AssertionError: expected '.user.endsWith\\0000A9 a.childNode{color:green;}.user.endsWith\\AE a.childNode{color:yellow;}.user.Q\\000026A a.childNode{color:purple;}' to equal '.user.endsWith\\0000A9  a.childNode{color:green;}.user.endsWith\\AE  a.childNode{color:yellow;}.user.Q\\000026A a.childNode{color:purple;}'
      + expected - actual

      -.user.endsWith\0000A9 a.childNode{color:green;}.user.endsWith\AE a.childNode{color:yellow;}.user.Q\000026A a.childNode{color:purple;}
      +.user.endsWith\0000A9  a.childNode{color:green;}.user.endsWith\AE  a.childNode{color:yellow;}.user.Q\000026A a.childNode{color:purple;}
      
      at Context.<anonymous> (test/Parser.js:166:10)
      at processImmediate (node:internal/timers:464:21)
```